tools/ir_subgraph.py

```diff
@@ -0,0 +1,279 @@
+    for layer in xml_dict["layers"]["layer"]:
+        node_name = layer["attributes"]["name"]
+        if node_name in node_names:
+            result_xml_dict["layers"]["layer"].append(layer)
+
+    # Keep only the edges that connect the visited nodes
+    result_xml_dict["edges"]["edge"] = []
+    for edge in xml_dict["edges"]["edge"]:
+        from_layer = int(edge["attributes"]["from-layer"])
+        to_layer = int(edge["attributes"]["to-layer"])
+        if from_layer in node_ids or to_layer in node_ids:
+            result_xml_dict["edges"]["edge"].append(edge)
+
+    return result_xml_dict
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Extract a subgraph from a model in OpenVINO Intermediate Representation format. Subgraph is taken "
+        "around a given node. Use distance parameter to control how many nodes around the given one to include. "
+        "The resulting subgraph is saved next to the input .xml file or at --output_path if provided. Additionally, a "
+        "symbolic link targeting the original .bin file is created."
+    )
+
+    parser.add_argument("input-path", help="Input IR path.")
+    parser.add_argument("node", help="Target node name.")
+    parser.add_argument("--distance", type=int, default=10, help="Distance around the target node (default 10).")
+    parser.add_argument(
+        "--output-path",
+        dest="output_path",
+        help="Output IR path. Can either be a file path with .xml extension or a directory path.",
+    )
+
+    args = parser.parse_args()
+
+    input_path = Path(args.__dict__["input-path"])
+    node_name = args.node
+    distance = args.distance
+    output_path = Path(args.output_path) if args.output_path is not None else None
+
+    if distance <= 0:
+        raise ValueError("Distance should be positive")
+
+    if output_path is None or output_path.suffix == "":
+        output_filename = f"{input_path.stem}_{Path(node_name).stem}_{distance}.xml"
+        if output_path is None:
+            output_dir = input_path.parent
+            output_path = input_path.parent / output_filename
+        else:
+            output_dir = output_path
+            output_path = output_dir / output_filename
+    else:
+        output_dir = output_path.parent
+
+    # Read IR xml as dict
+    tree = ET.parse(input_path)
+    root = tree.getroot()
+    xml_dict = xml_to_dict(root)
+
+    # Take subgraph
+    subgraph_xml_dict = take_model_subgraph(xml_dict, source_node_name=node_name, distance=distance)
+
+    # Save subgraph xml
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True)
+    write_xml(subgraph_xml_dict, output_path)
+
+    # Create a symbolic link to original .bin file
+    bin_output_path = output_path.with_suffix(".bin")
+    if bin_output_path.exists():
+        os.remove(bin_output_path)
+    bin_output_path.symlink_to(os.path.relpath(input_path.with_suffix(".bin"), bin_output_path.parent))
```

Comment:
Make sure that this works by default on Windows, otherwise we can't be using symlinks here.

Reply:
It works, but admin privileges are required for creating symlinks. Added a note for that.